### PR TITLE
[ETFE-2476] added DocumentReference controller/form/view

### PR DESCRIPTION
**Pull request by LadonJackson**

### CODE REVIEW

File: conf/documents.routes [hunk 1]
```diff
@@ -14,3 +14,13 @@ GET        /trader/:ern/draft/:draftId/documents/document-description
 POST       /trader/:ern/draft/:draftId/documents/document-description                        controllers.sections.documents.DocumentDescriptionController.onSubmit(ern: String, draftId: String, mode: Mode = NormalMode)
 GET        /trader/:ern/draft/:draftId/documents/document-description/change                 controllers.sections.documents.DocumentDescriptionController.onPageLoad(ern: String, draftId: String, mode: Mode = CheckMode)
 POST       /trader/:ern/draft/:draftId/documents/document-description/change                 controllers.sections.documents.DocumentDescriptionController.onSubmit(ern: String, draftId: String, mode: Mode = CheckMode)
+
+GET        /trader/:ern/draft/:draftId/documents/document-description                        controllers.sections.documents.DocumentDescriptionController.onPageLoad(ern: String, draftId: String, mode: Mode = NormalMode)
```
Comment (ericohwotu) [inline conf/documents.routes:18]: Duplicate description urls

File: conf/messages [hunk 2]
```diff
@@ -847,3 +847,19 @@ documentDescription.error.length = Document description must be 350 characters o
 documentDescription.error.xss = Document description must not contain < and > and : and ;
 documentDescription.change.hidden = documentDescription
 documentDescription.checkYourAnswersLabel = documentDescription
+
+documentDescription.title = Enter a description for this document
```
Comment (ericohwotu) [inline conf/messages:851]: duplicate documentDescription messages

File: test/navigation/DocumentsNavigatorSpec.scala [hunk 3]
```diff
@@ -36,44 +36,49 @@ class DocumentsNavigatorSpec extends SpecBase {
...
-        }
+        navigator.nextPage(DocumentsCertificatesPage, NormalMode, emptyUserAnswers) mustBe
+          testOnly.controllers.routes.UnderConstructionController.onPageLoad()
       }
 
-      "for the ReferenceAvailablePage" - {
+      "must go from ReferenceAvailablePage" - {
 
-        "must go to ReferenceAvailablePage when user selects yes" in {
+        "to ReferenceAvailablePage when user selects yes" in {
```
Comment (ericohwotu) [inline test/navigation/DocumentsNavigatorSpec.scala:48]: should this be DocumentReferencePage

File: test/navigation/DocumentsNavigatorSpec.scala [hunk 4]
```diff
@@ -36,44 +36,49 @@ class DocumentsNavigatorSpec extends SpecBase {
...
 
           val userAnswers = emptyUserAnswers.set(ReferenceAvailablePage, true)
 
           navigator.nextPage(ReferenceAvailablePage, NormalMode, userAnswers) mustBe
-            testOnly.controllers.routes.UnderConstructionController.onPageLoad()
+            routes.DocumentReferenceController.onPageLoad(testErn, testDraftId, NormalMode)
         }
 
-        "must go to ReferenceAvailablePage when user selects no" in {
+        "to ReferenceAvailablePage when user selects no" in {
```
Comment (ericohwotu) [inline test/navigation/DocumentsNavigatorSpec.scala:56]: think name is incorrect ReferenceAvailablePage

File: test/views/sections/documents/DocumentReferenceViewSpec.scala [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import play.api.i18n.Messages
+import play.api.mvc.AnyContentAsEmpty
+import play.api.test.FakeRequest
+import views.html.sections.documents.DocumentReferenceView
+import views.{BaseSelectors, ViewBehaviours}
+
+class DocumentReferenceViewSpec extends ViewSpecBase with ViewBehaviours {
+  object Selectors extends BaseSelectors
+
+  "Dispatch Business Name view" - {
```
Comment (ericohwotu) [inline test/views/sections/documents/DocumentReferenceViewSpec.scala:34]:
```suggestion
  "Document Reference View" - {
```

File: test/views/sections/documents/DocumentReferenceViewSpec.scala [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+    Seq(DocumentReferenceMessages.English).foreach { messagesForLanguage =>
```
Comment (ericohwotu) [inline test/views/sections/documents/DocumentReferenceViewSpec.scala:36]: [optional] we no longer need a seq here as its just on language

File: app/forms/sections/documents/DocumentReferenceFormProvider.scala [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import play.api.data.Form
+
+import javax.inject.Inject
+
+class DocumentReferenceFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[String] =
+    Form(
+      "value" -> text("documentReference.error.required")
+        .transform[String](
```
Comment (ericohwotu) [inline app/forms/sections/documents/DocumentReferenceFormProvider.scala:30]: This transformation looks really useful. 👍🏿 will it be worth having different `text()` e.g. `normalisedSpaceText()` which does this automatically?

**Review by nickg-hmrc — APPROVED** [review]

**Review by r-melvin — APPROVED** [review]
LGTM provided tests pass

**Pull request merged**